salt/utils/extend.py

```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+'''
+SaltStack Extend
+~~~~~~~~~~~~~~~~
+
+A templating tool for extending SaltStack.
+
+Takes a template directory and merges it into a SaltStack source code
+directory. This tool uses Jinja2 for templating.
+
+This tool is accessed using `salt-extend`
+
+    :codeauthor: :email:`Anthony Shaw <anthonyshaw@apache.org>`
+'''
+from __future__ import absolute_import
+from __future__ import print_function
+import yaml
+from collections import OrderedDict
+from datetime import date
+import tempfile
+import os
+import shutil
+from jinja2 import Template
+
+# zip compat for PY2/3
+from salt.ext.six.moves import zip
+
+import logging
+log = logging.getLogger('salt-extend')
+
+try:
+    import click
+    HAS_CLICK = True
+except ImportError as ie:
+    HAS_CLICK = False
+
+
+def _get_template(path, option_key):
+    """
+    Get the contents of a template file and provide it as a module type
+
+    :param path: path to the template.yml file
+    :type  path: ``str``
+
+    :param option_key: The unique key of this template
+    :type  option_key: ``str``
+
+    :returns: Details about the template
+    :rtype: ``tuple``
+    """
+    with open(path, "r") as template_f:
```

Comment:
Although salt-extend is not going to be used as much as all other salt code, replacing all `open()` calls with `salt.utils.fopen()` is good practice.